aws/rust-runtime/aws-config/src/profile/credentials/exec.rs

```diff
@@ -50,10 +54,25 @@ impl AssumeRoleProvider {
             .as_ref()
             .cloned()
             .unwrap_or_else(|| sts::util::default_session_name("assume-role-from-profile"));
+
+        let token = match &self.mfa_serial {
+            Some(mfa_serial) => {
+                client_config.mfa_token_provider
+                    .mfa_token(mfa_serial)
+                    .instrument(tracing::debug_span!("request_mfa_token"))
+                    .await
+                    .map(|t| Some(t.0.clone()))
```

Comment:
For standard virtual mfa devices, this token should be short (6 characters) - though that isn't explicitly checked yet. 

1. I think the string copy is okay - we can't get away from a string copy, since we pass it into `.set_token_code` below
2. I'm not sure whether / where tokens should be validated? I think non-virtual mfa devices can potentially have differently shaped codes, so a regex / 6-char check here might be too restrictive. The worst that can happen is that we push an invalid token to the auth backend (... which will be rejected with a reasonably intelligable error). I think perhaps validating this is the responsibility of the `ProvideMfaToken` implementation - but I'm open to suggsetions.

Reply:
Does boto3 do any validation? I think that unless we have some programmatic way to know the given constraints a particular MFA process requires, we can't validate this.

Reply:
Boto checks that tokens have >= 6 characters in them before sending the request, in the operator validation layer. It doesn't check that eg. the token is only digits or anything like that. Rust SDK doesn't seem to do any validation at all on the operation yet, but you do get a sensible error back from the API, so I think this is okay (or, something to punt to a more general "operator validation" improvement).